pkg/controller/infinispan/util/cluster.go

```diff
@@ -30,9 +30,13 @@ type ClusterInterface interface {
 	GracefulShutdown(secretName, podName, namespace, protocol string) error
 	GetClusterMembers(secretName, podName, namespace, protocol string) ([]string, error)
 	ExistsCache(cacheName, secretName, podName, namespace, protocol string) bool
+	ExistsCacheWithAuth(user, pass, cacheName, podName, namespace, protocol string) (bool, error)
 	CreateCache(cacheName, cacheXml, secretName, podName, namespace, protocol string) error
+	CreateCacheWithAuth(user, pass, cacheName, cacheXML, podName, namespace, protocol string) error
+	CreateCacheWithTemplateName(user, pass, cacheName, templateName, podName, namespace, protocol string) error
```

Comment:
@rigazilla maybe change name with common pattern (CreateCacheWith*), like CreateCacheWithTemplate
As we have already a large number of combinations create a common one function with all attrs, like user, pass and templateName and define logic inside depends of the provided values (nil or not nil)
wdyt?

Reply:
ok for the name change, but I would prefer to keep small separated methods with clear interface instead of having 1 big func with all the logic mixed inside.

Reply:
Ok, good point, let's keep your strategy as it can be easy to maintain

Reply:
since this impact method used in the infinispan reconciler I would do this in a next PR together with:
- removing methods that access secret
- removing hardcoded `operator` user